test/e2e/specs/treeview/00-static-treeview.spec.js

```diff
@@ -37,23 +37,132 @@ exports.tests = function (appName, baseUrl) {
           expect(searchDiv.isDisplayed()).toBeTruthy("Search box input is not visible when page loads initially.");
         });
 
-        it('expand all and collapse all button are visible', function(){
+      });
+
+      describe('actions for the page', function() {
+        beforeAll(function() {
+          browser.wait(EC.elementToBeClickable(filterDropDown), browser.params.defaultTimeout);
+          filterDropDown.click();
+          browser.wait(EC.visibilityOf(element(by.id('number-menu'))), browser.params.defaultTimeout)
+          element(by.id('ui-id-2')).click();
+
+          deriva.waitForElementInverse(element.all(by.id("loadIcon")).get(0));
```

Comment:
You should use chained promises for click events as well so that the control flow will work properly. Each `browser.wait()` and each `.click()` push a promise into the queue to be executed at a later time. The way you wrote this test won't cause any problems because each wait/click condition will be acted on in the same order they were pushed.

```js
browser.wait(EC.elementToBeClickable(filterDropDown), browser.params.defaultTimeout);
filterDropDown.click().then(function () {
  browser.wait(EC.visibilityOf(element(by.id('number-menu'))), browser.params.defaultTimeout)
  return element(by.id('ui-id-2')).click();
}).then(function () {
  deriva.waitForElementInverse(element.all(by.id("loadIcon")).get(0));
}).catch(...);
```

This chaining will ensure that the click events are triggered when we want them to be. If you want to be more explicit, then we can add the wait commands to the promise chain as well:
```js
browser.wait(EC.elementToBeClickable(filterDropDown), browser.params.defaultTimeout).then(function () {
  return filterDropDown.click()
}).then(function () {
  return browser.wait(EC.visibilityOf(element(by.id('number-menu'))), browser.params.defaultTimeout);
}).then(function () {
  return element(by.id('ui-id-2')).click();
}).then(function () {
  deriva.waitForElementInverse(element.all(by.id("loadIcon")).get(0));
}).catch(...);
```
This change doesn't need to be done, but it uses the control flow how each of these functions intend it to be used. Most events that are done with protractor will emit a promise that _can_ be acted on. 

The use of `done` allows us to be even more explicit as to when the test should notify the next test to start running. This can also be used in a `beforeAll` function:
```js
beforeAll(function(done) {
  browser.wait(EC.elementToBeClickable(filterDropDown), browser.params.defaultTimeout).then(function () {
    return filterDropDown.click()
  }).then(function () {
    return browser.wait(EC.visibilityOf(element(by.id('number-menu'))), browser.params.defaultTimeout);
  }).then(function () {
    return element(by.id('ui-id-2')).click();
  }).then(function () {
    return deriva.waitForElementInverse(element.all(by.id("loadIcon")).get(0));
  }).then(function () {
    done();
  }).catch(function (err) {
    console.log(err);
    done.fail();
  });
```